tds/db/graph/search_provenance.py

```diff
@@ -12,38 +26,47 @@ class SearchProvenance(ProvenanceHandler):
     def __init__(self, rdb, graph_db):
         super().__init__(rdb=rdb, graph_db=graph_db)
 
-    def dynamic_relationship_direction(self, direction):
-        """
-        get direction of relationship based on direction type
-        """
-        if direction == "all":
-            return "-[*]-"
-        if direction == "child":
-            return "<-[*]-"
-        if direction == "parent":
-            return "-[*]->"
-        raise Exception("relationship direction is not allowed.")
+    def __getitem__(self, key):
+        return self.__getattribute__(key)
 
     def connected_nodes_by_direction(self, payload, direction):
         """
         Connect nodes
         """
         with self.graph_db.session() as session:
+            # return string of relationship excluding CONTAINS and IS_CONCEPT_OF
+            relationships_str = relationships_array_as_str(
+                exclude=["CONTAINS", "IS_CONCEPT_OF"]
+            )
+
+            # set the direction of the search dynamically
+            relation_direction = dynamic_relationship_direction(
+                direction=direction, relationship_type=f"r:{relationships_str} *1.."
+            )
+
+            # build the first match node
+            match_node = match_node_builder(
+                node_type=payload.get("root_type"), node_id=payload.get("root_id")
+            )
 
+            # build the query. n is an arbitrary node
             query = (
-                f"Match (n1: {payload.get('root_type')}) "
-                + f"{self.dynamic_relationship_direction(direction=direction)}(n2)"
-                + "Where n1.id = $root_id "
-                + "RETURN labels(n2) as label, n2.id as id"
+                f"{match_node}"
+                + f"{relation_direction}(n) "
+                + "With DISTINCT n "
+                + "RETURN labels(n) as label, n.id as id"
             )
+            print(query)
+            logging.info(query)
 
-            response = session.run(query, root_id=payload.get("root_id"))
+            response = session.run(query)
 
-            return [
-                {"label": res.data().get("label")[0], "id": res.data().get("id")}
-                for res in response
+            response_data = [
+                {res.data().get("label")[0]: res.data().get("id")} for res in response
```

Comment:
What is `res`? Is it okay if this line is broken up. Just not sure what it's doing.

Reply:
res is just part of the result that is looping. It is going to be changed anyways in my next pr to account for returning the whole graph.